docs/parameters.md

```diff
@@ -0,0 +1,150 @@
+
+There are multiple use cases from the users perspective that dictate what parameters need to be available for user input.
+
+ * The user has deployed an external database and needs to specify connection details such as host and port.
+ * The user wishes to manage passwords externally to the installer and needs to specify rather than use the default randomly generated password.
+ * The user has deployed an external database configured with TLS and needs to specify the CA certificate.
+ * The user has deployed an external database configured with TLS and certificate authentication.
+
+##### Mapped
+
+| Parameter | Description | foreman-installer Parameters |
+| ----------| ----------- | ---------------------------- |
+| `--database-host` | Location to connect to the database | `--foreman-db-host`<br/> `--katello-candlepin-db-host`<br/> `--foreman-proxy-content-pulpcore-postgresql-host` |
+| `--database-port` | Port to connect to the database | `--foreman-db-port`<br/> `--katello-candlepin-db-port`<br/> `--foreman-proxy-content-pulpcore-postgresql-port` |
+| `--database-password` | Password for each database | `--foreman-db-password`<br/> `--katello-candlepin-db-password`<br/> `--foreman-proxy-content-pulpcore-postgresql-password` |
+| `--database-ssl-mode` | SSL verification mode to use | `--foreman-db-sslmode` <br/> `--katello-candlepin-db-ssl-verify` |
+| `--database-ssl-ca` | Path to the database CA certificate | `--foreman-db-root-cert` <br/> `--katello-candlepin-db-ssl-ca` <br/> `--foreman-proxy-content-pulpcore-db-ssl-root-ca` |
+| `--database-ssl-cert` | Path to the database certificate | |
+| `--database-ssl-key` | Path to the database private key | |
+
+##### Unmapped
+
+| foreman-installer Parameter | Description | Reason |
+| --------------------------- | ----------- | ------ |
+| `--katello-candlepin-manage-db` | When true, creates a local database | ?? |
+| `--katello-candlepin-db-name` | Sets the name of the Candlepin database | The user should not configure this and instead use a hard-coded name |
+| `--katello-candlepin-db-user` | Sets the user of the Candlepin database | The user should not configure this and instead use a hard-coded name |
+| `--katello-candlepin-db-ssl` | Sets whether database connection is over SSL | The SSL mode parameter can be used to determine this |
+| `--foreman-proxy-content-pulpcore-manage-postgresql` | When true, creates a local database | ?? |
+| `--foreman-proxy-content-pulpcore-postgresql-db-name` | Sets the name of the Pulpcore database | The user should not configure this and instead use a hard-coded name |
+| `--foreman-proxy-content-pulpcore-postgresql-user` | Sets the user of the Pulpcore database | The user should not configure this and instead use a hard-coded name |
+| `--foreman-proxy-content-pulpcore-postgresql-ssl` | Sets whether database connection is over SSL | The SSL mode parameter can be used to determine this |
+| `--foreman-db-manage` | When true, creates a local database | ?? |
```

Comment:
how will the user decide whether the DB is managed by the installer or external? should there be a `--database-mode [internal|external]` switch?

Reply:
Great question, I left that here to spark the debate before just assuming. I wondered if we could trigger this off of "is database_host == localhost" but I suppose a dedicated parameter would allow us better user input requirement handling at the CLI level?

Reply:
We had a similar discussion around certificate handling and ended at "explicit is better", so let's be explicit